lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -769,7 +769,12 @@ def _maybe_depth_shade_and_sort_colors(self, color_array):
         )
         if len(color_array) > 1:
             color_array = color_array[self._z_markers_idx]
-        return mcolors.to_rgba_array(color_array, self._alpha)
+
+        if self._alpha is not None:
+            converted_alphas = color_array[:, 3] * self._alpha
```

Comment:
Are we sure this will never get an RGB (vs RGBA) array?

Reply:
Hi @tacaswell. Good point, if depthshape is specified as False a RGBA type array might get passed. I put an escape clause for this case.
```suggestion
            return mcolors.to_rgba_array(color_array, self._alpha)
```